blocks/library/audio/index.js

```diff
@@ -20,7 +20,7 @@ import Editable from '../../editable';
 import BlockControls from '../../block-controls';
 import BlockAlignmentToolbar from '../../block-alignment-toolbar';
 
-registerBlockType( 'core/audio', {
+export const registerAudioBlock = () => registerBlockType( 'core/audio', {
```

Comment:
This seems very verbose.

Reply:
```js
export default () => registerBlockType( 'core/audio', {
```

is another way to go.

Reply:
Either that or
```js
export const Audio = () => registerBlockType( 'core/audio', {} );
```

Reply:
What about exporting the name and the settings and loop over the blocks and call `registerBlockType` when needed? I think we explored this possibility at some point can't recall why we dismissed it

Reply:
@youknowriad can you give an example? How would it work with embeds file which contains dozen definitions?

Reply:
```
import * as audio from 'audio';
import embeds from 'embed';

const blockConfigs = [
 audio,
 ...embeds
];

blockConfigs.forEach( ( { name, settings } ) => registerBlockType( name, settings ) );
```

The embeds woud export an array `[{ name, settings }]`, while the others would export `{ name, settings }`. I don't know if this is a good option though.

I guess the main advantage of this approach is the possibility to test the settings without registering the blocks.

Reply:
Interesting, @mtias & @aduth what do you think? I'm fine with that, too.

Reply:
https://github.com/WordPress/gutenberg/pull/4514/commits/2c00ffcf51bcebe83641765b2729f4e4418c3eec - I like `name` & `settings` pair. It gives us more flexibility when testing code.

Reply:
Makes me wonder if `name` and `settings` need to be separate at all, e.g.

```js
export default {
	name: 'core/foo',

	// ...settings
};
```

We could overload `registerBlockType` to support both this and the current implementation.

Thinking back to some of the original reasons for the API as it stands:

- Compatibility with similar APIs like [`register_post_type`](https://codex.wordpress.org/Function_Reference/register_post_type) and [`register_rest_route`](https://developer.wordpress.org/reference/functions/register_rest_route/)†
- More easy to identify name at a glance (also enforcing as _first_, which is not true as an object property)

To such API design generally, the `string, object` argument signature seems best-suited in cases where the object contains properties none of which are necessarily required. An ordered set of arguments is good when all arguments are required and the set is limited in size (since otherwise the ordering loses its obviousness on what each value represents).

Here, it's a bit of an awkward mismatch. The name is required, but so are some of the options (e.g. `save`). Conversely, not all block properties are required (e.g. `supports`). To me, other APIs notwithstanding, the single object argument seems the most sensible.

`</braindump>`

_† But this could serve as a counterpoint, since unlike in `register_rest_route`, we don't separate the namespace and block name arguments._